src/AvRadio.js

```diff
@@ -0,0 +1,57 @@
+import React, {PropTypes} from 'react';
+import classNames from 'classnames';
+import {Input, FormGroup, Label} from 'reactstrap';
+import AvInput from './AvInput'
+
+let radioPropTypes = Object.assign({}, AvInput.propTypes);
+delete radioPropTypes.name;
+
+export default class AvRadio extends AvInput {
+
+  static contextTypes = Object.assign({}, AvInput.contextTypes, {
+    Group: PropTypes.object.isRequired,
+  });
+
+  static propTypes = radioPropTypes;
+
+  constructor (props) {
+    super(props);
+
+    this.onChangeHandler = ::this.onChangeHandler;
+  }
+
+  onChangeHandler () {
+    this.context.Group.update(this.props.value);
+  }
+
+  render () {
+    const {errorMessage, validate, validationEvent, state, className, tag: Tag, ...attributes} = this.props;
+
+    const classes = classNames(
+      className,
+      this.context.FormCtrl.isTouched[this.props.name] ? 'av-touched' : 'av-untouched',
+      this.context.FormCtrl.isDirty[this.props.name] ? 'av-dirty' : 'av-pristine',
+      this.context.FormCtrl.isBad[this.props.name] ? 'av-bad-input' : null,
+      this.context.FormCtrl.hasError[this.props.name] ? 'av-invalid' : 'av-valid'
+    );
+
+    const isDisabled = attributes.disabled === true;
```

Comment:
ok, I switched it to use it directly, but at the moment we're using it on both the formgroup and on the input.